src/terrain/terrain.cpp

```diff
@@ -918,15 +913,35 @@ Terrain::get_path(
             }
         }
     }
-    std::vector<const T*> out;
-    return out;
+    std::vector<const T*> path;
+    return path;
+}
+
+uint32_t
+Terrain::get_voxel(int x, int y, int z) const {
+    // using static ints to prevent dereferencing
+    static unsigned int get_voxel_mat_id = 0;
+    static unsigned int get_voxel_color_id = 0;
+    static uint32_t get_voxel_out_color = 0;
+    if (in_range(x, y, z)) {
+        if ((tiles[pos(x, y, z)].get_material_id() != get_voxel_mat_id)
+            | (tiles[pos(x, y, z)].get_color_id() != get_voxel_color_id)) {
+            get_voxel_mat_id = tiles[pos(x, y, z)].get_material_id();
+            get_voxel_color_id = tiles[pos(x, y, z)].get_color_id();
+            auto mat = materials_->at(get_voxel_mat_id);
+            get_voxel_out_color = mat.color[get_voxel_color_id].second;
```

Comment:
Not formatted correctly

Reply:
according to your linter it is

Reply:
Linter checks for bad practices in coding (i.e. UB checks)
Formatter makes all code look the same

Linters check function
Formatters check form

Reply:
Im so sorry,
according to your formater it is